llpc/context/llpcPipelineContext.cpp

```diff
@@ -420,10 +420,13 @@ void PipelineContext::setOptionsInPipeline(Pipeline *pipeline, Util::MetroHash64
     shaderOptions.allowReZ = shaderInfo->options.allowReZ;
     shaderOptions.forceLateZ = shaderInfo->options.forceLateZ;
 
-    if (shaderInfo->options.vgprLimit != 0 && shaderInfo->options.vgprLimit != UINT_MAX)
-      shaderOptions.vgprLimit = shaderInfo->options.vgprLimit;
-    else
-      shaderOptions.vgprLimit = VgprLimit;
+    shaderOptions.vgprLimit = shaderInfo->options.vgprLimit;
```

Comment:
This changed the behavior compared to before.
Before, in the case that `shaderInfo->options.vgprLimit == UINT_MAX`, `shaderOptions.vgprLimit = 0` was set (when there was no cli option), now, `shaderOptions.vgprLimit = UINT_MAX` is set, which produces warnings in the backend:
```
ERROR: LLVM DIAGNOSIS INFO: can't parse integer attribute amdgpu-num-vgpr
```